sim-lib/src/lib.rs

```diff
@@ -1,44 +1,87 @@
 use bitcoin::secp256k1::PublicKey;
 use lightning::ln::PaymentHash;
-use serde::{Deserialize, Serialize};
-use std::time::SystemTime;
+use serde::{de::Visitor, Deserialize, Deserializer, Serialize, Serializer};
+use std::{collections::HashMap, str::FromStr, sync::Arc, time::SystemTime};
+
+pub mod lnd;
 
 // Phase 0: User input - see config.json
 
 // Phase 1: Parsed User Input
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct NodeConnection {
-    id: String,
-    host: String,
-    macaroon: String,
-    cert: String,
+    pub id: String,
+    pub address: String,
+    pub macaroon: String,
+    pub cert: String,
 }
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct Config {
-    nodes: Vec<NodeConnection>,
+    pub nodes: Vec<NodeConnection>,
+    pub activity: Vec<ActivityDefinition>,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ActivityDefinition {
     // The source of the action.
-    source: PublicKey,
+    #[serde(
+        serialize_with = "serialize_public_key",
+        deserialize_with = "deserialize_public_key"
+    )]
+    pub source: PublicKey,
     // The destination of the action.
-    dest: PublicKey,
+    #[serde(
+        serialize_with = "serialize_public_key",
+        deserialize_with = "deserialize_public_key"
+    )]
+    pub dest: PublicKey,
     // The frequency of the action, as in number of times per minute.
-    frequency: u16,
+    pub frequency: u16,
     // The amount of m_sat to used in this action.
-    amt_msat: u64,
+    pub amt_msat: u64,
+}
+
+fn serialize_public_key<S>(key: &PublicKey, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    key.to_string().serialize(serializer)
+}
+
+struct PublicKeyVisitor;
+
+impl<'de> Visitor<'de> for PublicKeyVisitor {
+    type Value = PublicKey;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str("a hex string representing a public key")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        PublicKey::from_str(value).map_err(serde::de::Error::custom)
+    }
+}
+
+fn deserialize_public_key<'de, D>(deserializer: D) -> Result<PublicKey, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    deserializer.deserialize_str(PublicKeyVisitor)
```

Comment:
I feel like this should not be needed, but let's go with it for now and try to simplify it later

Reply:
Ok, just add  `features = [ "serde" ]` to the bitcoin crate in `Cargo.toml`